specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

```diff
@@ -3856,6 +3856,37 @@
         "dataset"
       ]
     },
+    "AzureDataExplorerCommand": {
+      "x-ms-discriminator-value": "AzureDataExplorerCommand",
+      "description": "Azure Data Explorer command activity.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ExecutionActivity"
+        }
+      ],
+      "properties": {
+        "typeProperties": {
+          "description": "Azure Data Explorer command activity properties.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AzureDataExplorerCommandActivityTypeProperties"
+        }
+      },
+      "required": [
+        "typeProperties"
+      ]
+    },
+    "AzureDataExplorerCommandActivityTypeProperties": {
+      "description": "Azure Data Explorer command activity properties.",
+      "properties": {
+        "command": {
+          "type": "object",
+          "description": "A control command, according to the ADX command syntax. Type: string (or Expression with resultType string)."
```

Comment:
is the `command` property a string or object?

Reply:
It is eventually a string, but also accepts an Expression which is compiled into a string in the backend. Expression is another type and since we need both to be accepted here (oneOf model from swagger 3.0) we use object type in all these type of properties.